api/provision.go

```diff
@@ -0,0 +1,83 @@
+package api
+
+import (
+	"encoding/json"
+)
+
+type VMConfig struct {
+	CPU    int
+	Memory int
+	Kernel string
+	Initrd string // it can be nil, boot image should be added via AddDevice() in this case
+	//Bios   string
+	//Cbfs   string
+
+	CPUMax        int
+	MemSlot       int
+	MemMax        int
+	MemPath       string
+	MemPathShared bool
+
+	// to be filled
+}
+
+type QosConfig struct {
+	CgroupPath string
+
+	// For network QoS (kilobytes/s)
+	InboundAverage  string
+	InboundPeak     string
+	OutboundAverage string
+	OutboundPeak    string
+
+	// to be filled
+}
```

Comment:
virtcontainers does not have this kind of setting, that is a potential gap. Could you please explain where that is used and what it's for?